spec/domain/messages/bulk_mail_dispatch_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2021, CVP Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+require 'spec_helper'
+
+describe Messages::BulkMailDispatch do
+  let(:message) { messages(:mail) }
+  let(:top_leader) { people(:top_leader) }
+  let(:dispatch) { described_class.new(message) }
+
+  let(:address_list) {
+    [
+      { person_id: top_leader.id, email: 'recipient1@example.com' },
+      { person_id: top_leader.id, email: 'recipient2@example.com' }
+    ]
+  }
+
+  let(:recipients) {
+    [
+      MessageRecipient.new(message_id: message.id, person_id: top_leader.id, state: :pending, email: 'recipient1@example.com'),
+      MessageRecipient.new(message_id: message.id, person_id: top_leader.id, state: :pending, email: 'recipient2@example.com')
+    ]
+  }
+
+  before do
+    allow(Truemail).to receive(:valid?).and_call_original
+    Subscription.create!(mailing_list: mailing_lists(:leaders), subscriber: top_leader)
+  end
+
+  context '#run' do
+    it 'delivers mail if message recipients do exist' do
+      message.message_recipients = recipients
```

Comment:
create message_recipient in db with create!
this assures that entries are fetched from db. 
remove let block further up if not required anymore.